olive/engine/engine.py

```diff
@@ -497,40 +488,60 @@ def resolve_goals(
         goals = {}
         multipliers = {}
         for metric in metrics:
-            if metric.goal is not None:
-                goals[metric.name] = metric.goal
-                multipliers[metric.name] = 1 if metric.higher_is_better else -1
-        if verbose and len(goals) > 0:
+            # only resolve sub metrics whose priority > 0
+            goals[metric.name] = metric.get_sub_type_info("goal")
+            multipliers[metric.name] = metric.get_sub_type_info(
+                info_name="higher_is_better",
+                callback=lambda x: 1 if x else -1,
+            )
+
+        if verbose and goals:
             logger.info(f"Resolving goals: {goals}")
 
-        # compute baseline for input model if needed
-        baseline = {}
-        for _, goal in goals.items():
-            if goal.type != "threshold":
-                assert self.evaluator_config is not None, "Default evaluator must be provided to resolve goals"
-                if verbose:
-                    logger.info("Computing baseline for metrics ...")
-                baseline = self._evaluate_model(
-                    input_model, input_model_id, self.evaluator_config, accelerator_spec, verbose=False
-                )
+        baseline = MetricResult()
+        for goal in goals.values():
+            _evaluated = False
+            for sub_goal in goal.values():
+                if not sub_goal:
+                    break
+                if sub_goal.type != "threshold":
+                    assert self.evaluator_config is not None, "Default evaluator must be provided to resolve goals"
+                    if verbose:
+                        logger.info("Computing baseline for metrics ...")
```

Comment:
could we use logger.debug without checking the verbose flag?

Reply:
Good suggestion. How about to fix in separated PR. The verbose flag is added in the first version of olive by Jambay.

Reply:
sure, for existing code, I suggest to fix in another PR, I put comments here only for new added code.

Reply:
removed.